src/coreclr/inc/CrstTypes.def

```diff
@@ -201,6 +201,10 @@ End
 Crst Exception
 End
 
+Crst ExecutableAllocatorLock
+    Unordered
```

Comment:
Is it really safe to have it unordered? Unordered means prone to deadlocks.

Reply:
I haven't found a way to define the lock so that it doesn't hit the ordering asserts. If I just removed the Unordered, then I get e.g. this assert:

  Assert failure(PID 12572 [0x0000311c], Thread: 51440 [0xc8f0]): Consistency check failed: Crst Level violation: Can't take level 0 lock CrstExecutableAllocatorLock because you already holding level 0 lock CrstLoaderHeap

All I want here is a leaf lock. The methods in the allocator can be called from all kinds of places, so I don't know what locks it is going to be called under.

Reply:
You can use `CrstLeafLock`. Or you can add `AcquiredAfter CrstLoaderHeap` for `CrstExecutableAllocatorLock`

Reply:
LeafLock doesn't work, with it it still complains with the same assert. Fortunately, the AcquiredAfter LoaderHeap seems to work, so I'll use that one.